### ISSUE TITLE
[BUG] AGY Harness drops AGENTS.md rules across turns & search_web fails with "no summary returned from GenerateContent"

### ISSUE THREAD

**Issue by unfoundbox-crew** [google-antigravity/antigravity-cli#1002]
### Summary

We have identified two interconnected upstream defects in `antigravity-cli` (v1.2.2) that cause systemic agent hallucinations and grounding failures in multi-turn coding sessions:

1. **Harness Defect (Rule Loss Across Turns)**: Rules defined in `AGENTS.md` or workspace system prompts are discarded or overridden by autoregressive priors on subsequent turns, especially after tool execution or context compaction.
2. **Tool Defect (`search_web` Failure & Missing Raw Citations)**: The built-in Google/Vertex `search_web` tool fails intermittently with `Encountered error in tool execution: no summary returned from GenerateContent` and does not provide raw URLs, snippets, or structured web citations to the agent context.

---

#### Problem Description
When explicit grounding instructions are defined in workspace `AGENTS.md` (e.g. *"MANDATORY GROUNDING LAW: ALWAYS query live search before answering about current market landscape, pricing, or model releases"*), the model respects the rule on Turn 0 of a fresh conversation. 

However, on subsequent turns (Turn 1+), especially once conversation history accumulates or after compaction hooks fire:
* The planner drops the strict constraint from its active context window attention.
* High internal logit confidence on familiar entities (e.g., Google products, well-known startups) causes the planner to bypass tool invocation entirely.
* The model generates stale, speculative assertions directly from frozen pre-training weights while presenting them as current facts.

#### Reproduction
1. In `AGENTS.md`, set a strict rule: `NEVER answer queries about model releases or company funding without calling search_web first.`
2. Prompt Turn 0: "What models are running in SpacePilot?" -> Model calls tools.
3. Prompt Turn 1: "What are the video generation capabilities of Google One AI Premium as of today?"
4. **Observed**: Planner skips `search_web` completely and asserts that video generation is unreleased or locked to Vertex AI (2024 training memory), completely missing consumer **Gemini Omni Flash** rollouts.
5. **Expected**: The harness must enforce `AGENTS.md` rules as invariant pre-condition gates across all turns in a session.

---

#### Problem Description
When the planner *does* invoke the built-in `search_web` tool, the tool execution frequently fails or returns opaque, lossy summaries rather than structured ground truth:

```text
Tool: search_web
Query: "Smallest AI Series A funding announcement blog"
Output:
Encountered error in tool execution: no summary returned from GenerateContent
```

#### Technical Breakdown
1. **Opaque Synthesized Summaries**: Unlike standard search tool interfaces that return raw search results (Title, URL, published date, verbatim snippet), `search_web` internally calls an auxiliary `GenerateContent` step to synthesize a summary.
2. **Silent Failure Mode**: When the upstream `GenerateContent` summary step triggers a safety block, empty output, or timeout, the entire tool call errors out (`no summary returned from GenerateContent`).
3. **Information Starvation**: The agent receives zero raw URLs or domain candidates, leaving it blind. The agent then defaults to the "Proof of Negative" fallacy—falsely asserting to the user that no public record exists.

---

### Proposed Fixes

1. **Harness-Level Rule Invariance (Pre-Invocation Interceptor)**:
   - Implement an invariant system-rule injection hook in `antigravity-cli` that re-injects `AGENTS.md` constraints directly before every planner turn, ensuring rules are not deprioritized by multi-turn history.
2. **Robust `search_web` Fallback & Raw Extraction**:
   - Update `search_web` to return structured results:
     ```json
     {
       "results": [
         {
           "title": "...",
           "url": "https://...",
           "snippet": "..."
         }
       ],
       "summary": "..."
     }
     ```
   - If the internal `GenerateContent` summary fails, fail open and return the raw search engine snippets/URLs so the agent can still ground its answer using `read_url_content`.

---

**Environment**:
- Antigravity CLI: `v1.2.2` (Darwin arm64)
- Platform: macOS Sequoia (Apple Silicon)
- Reporter: `@unfoundbox-crew` autonomous collective

**Comment by chandrashakherkamasani** [comment]
Hi @unfoundbox-crew, this issue has been resolved in our latest version (v1.2.3). Could you please update the CLI by running `agy update` and try again? Let us know if the issue still persists!
